opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -239,12 +239,136 @@ impl opentelemetry::logs::Logger for Logger {
 
 #[cfg(test)]
 mod tests {
+    use crate::resource::{
+        SERVICE_NAME, TELEMETRY_SDK_LANGUAGE, TELEMETRY_SDK_NAME, TELEMETRY_SDK_VERSION,
+    };
+    use crate::Resource;
+
     use super::*;
     use opentelemetry::global::{logger, set_logger_provider, shutdown_logger_provider};
     use opentelemetry::logs::Logger;
+    use opentelemetry::{Key, KeyValue, Value};
     use std::sync::{Arc, Mutex};
     use std::thread;
 
+    #[test]
+    fn test_logger_provider_default_resource() {
+        let assert_resource = |provider: &super::LoggerProvider,
+                               resource_key: &'static str,
+                               expect: Option<&'static str>| {
+            assert_eq!(
+                provider
+                    .config()
+                    .resource
+                    .get(Key::from_static_str(resource_key))
+                    .map(|v| v.to_string()),
+                expect.map(|s| s.to_string())
+            );
+        };
+        let assert_telemetry_resource = |provider: &super::LoggerProvider| {
+            assert_eq!(
+                provider
+                    .config()
+                    .resource
+                    .get(TELEMETRY_SDK_LANGUAGE.into()),
+                Some(Value::from("rust"))
+            );
+            assert_eq!(
+                provider.config().resource.get(TELEMETRY_SDK_NAME.into()),
+                Some(Value::from("opentelemetry"))
+            );
+            assert_eq!(
+                provider.config().resource.get(TELEMETRY_SDK_VERSION.into()),
+                Some(Value::from(env!("CARGO_PKG_VERSION")))
+            );
+        };
+
+        // If users didn't provide a resource and there isn't a env var set. Use default one.
+        let default_config_provider = super::LoggerProvider::builder().build();
+        assert_resource(
+            &default_config_provider,
+            SERVICE_NAME,
+            Some("unknown_service"),
+        );
+        assert_telemetry_resource(&default_config_provider);
+
+        // If user provided a resource, use that.
+        let custom_config_provider = super::LoggerProvider::builder()
+            .with_config(Config {
+                resource: Cow::Owned(Resource::new(vec![KeyValue::new(
+                    SERVICE_NAME,
+                    "test_service",
+                )])),
+            })
+            .build();
+        assert_resource(&custom_config_provider, SERVICE_NAME, Some("test_service"));
```

Comment:
could we assert that telemetry_resource is *not* present in this case?

Reply:
Definitely, added in https://github.com/open-telemetry/opentelemetry-rust/pull/1646/commits/74ae6c512a7dace053c01053393bd369743a7c8c